File: Assemblies/ProteanCMS/CMS/Cart/Subscriptions.JSONActions.vb
Option Strict Off
Option Explicit On
Imports System.Xml
Imports System.Collections
Imports System.Web.Configuration
Imports System.Data.SqlClient
Imports System.Web.HttpUtility
Imports VB = Microsoft.VisualBasic
Imports System.IO
Imports Protean.Tools.Xml
Imports Protean.Tools.Xml.XmlNodeState
Imports System
Imports System.Text
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Linq
Imports System.Collections.Generic


Partial Public Class Cms

    Partial Public Class Subscriptions

#Region "JSON Actions"

        Public Class JSONActions
            Public Event OnError(ByVal sender As Object, ByVal e As Protean.Tools.Errors.ErrorEventArgs)
            Private Const mcModuleName As String = "Protean.Subscription.JSONActions"
            Private Const cContactType As String = "Venue"
            Private myWeb As Protean.Cms
            Private myCart As Protean.Cms.Cart

            Public Sub New()
                Dim ctest As String = "this constructor is being hit" 'for testing
                myWeb = New Protean.Cms()
                myWeb.InitializeVariables()
                myWeb.Open()
                myCart = New Protean.Cms.Cart(myWeb)

            End Sub


            Public Function SubscriptionsProcess(ByRef myApi As Protean.API, ByRef jObj As Newtonsoft.Json.Linq.JObject) As String
                Try
                    Dim cProcessInfo As String = ""

                    'does nothing yet

                Catch ex As Exception
                    RaiseEvent OnError(Me, New Protean.Tools.Errors.ErrorEventArgs(mcModuleName, "GetCart", ex, ""))
                    Return ex.Message
                End Try
            End Function


        End Class

#End Region
    End Class

End Class

File: Assemblies/Protean.CMS/features/subscriptions/Subscriptions.JSONActions.cs
using System;
using Newtonsoft.Json.Linq;

namespace Protean
{


    public partial class Cms
    {

        public partial class Subscriptions
        {

            #region JSON Actions

            public class JSONActions
            {
                public event OnErrorEventHandler OnError;

                public delegate void OnErrorEventHandler(object sender, Tools.Errors.ErrorEventArgs e);
                private const string mcModuleName = "Protean.Subscription.JSONActions";
                private const string cContactType = "Venue";
                private Cms myWeb;
                private Cart myCart;

                public JSONActions()
                {
                    string ctest = "this constructor is being hit"; // for testing
                    myWeb = new Cms();
                    myWeb.InitializeVariables();
                    myWeb.Open();
                    myCart = new Cart(ref myWeb);

                }


                public string SubscriptionsProcess(ref Protean.Rest myApi, ref JObject jObj)
                {
                    try
                    {
                        string cProcessInfo = "";

                        // does nothing yet
                        return null;
                    }
                    catch (Exception ex)
                    {
                        OnError?.Invoke(this, new Tools.Errors.ErrorEventArgs(mcModuleName, "GetCart", ex, ""));
                        return ex.Message;
                    }


                }


            }

            #endregion
        }

    }
}